test/e2e/utils/api/expect.js

```diff
@@ -1,10 +1,29 @@
 import { $, text as textTaiko } from 'taiko'
 
-export const expectExists = async ({ text = null, selector = null }) => {
-  if (!text && !selector) throw new Error('One between text and selector is required')
+export const getSelection = async ({ text = null, selector = null, itemSelector = null }) => {
+  if (!text && !selector && !itemSelector) throw new Error('One between text, selector and itemSelector is required')
 
-  const element = text ? textTaiko(text) : $(selector)
+  const element = await (itemSelector || (text ? textTaiko(text) : $(selector)))
+
+  return element
+}
+
+export const expectExists = async ({ text = null, selector = null, itemSelector = null }) => {
+  const element = await getSelection({ text, selector, itemSelector })
 
   const exists = await element.exists()
   await expect(exists).toBeTruthy()
 }
+
+export const expectExistsExactlyNumberOfTimes = async ({
+  text = null,
+  selector = null,
+  itemSelector = null,
+  numberOfItems = 0,
```

Comment:
pass numberOfItems to expectExists. better not to have another method.
then default is 1. if you expect an item, at least it should be 1.
then update expectExists with await expect(items.length).toBe(numberOfItems)

Reply:
changed